gdpr/vendorlist-fetching_test.go

```diff
@@ -7,233 +7,697 @@ import (
+	testCases := []test{
+		{
+			description: "Fetch - No Fallback - Vendor List 1",
+			setup: testSetup{
+				enableTCF1Fetch:    true,
+				enableTCF1Fallback: false,
+				vendorListVersion:  1,
+			},
+			expected: vendorList1Expected,
+		},
+		{
+			description: "Fetch - No Fallback - Vendor List 2",
+			setup: testSetup{
+				enableTCF1Fetch:    true,
+				enableTCF1Fallback: false,
+				vendorListVersion:  2,
+			},
+			expected: vendorList2Expected,
+		},
+		{
+			description: "Fetch - Fallback - Vendor List 1",
+			setup: testSetup{
+				enableTCF1Fetch:    true,
+				enableTCF1Fallback: true,
+				vendorListVersion:  1,
+			},
+			expected: vendorList1Expected,
+		},
+		{
+			description: "Fetch - Fallback - Vendor List 2",
+			setup: testSetup{
+				enableTCF1Fetch:    true,
+				enableTCF1Fallback: true,
+				vendorListVersion:  2,
+			},
+			expected: vendorList2Expected,
+		},
+		{
+			description: "No Fetch - Fallback - Vendor List 1",
+			setup: testSetup{
+				enableTCF1Fetch:    false,
+				enableTCF1Fallback: true,
+				vendorListVersion:  1,
+			},
+			expected: vendorListFallbackExpected,
+		},
+		{
+			description: "No Fetch - Fallback - Vendor List 2",
+			setup: testSetup{
+				enableTCF1Fetch:    false,
+				enableTCF1Fallback: true,
+				vendorListVersion:  2,
+			},
+			expected: vendorListFallbackExpected,
+		},
+		{
+			description: "No Fetch - No Fallback - Vendor List 1",
+			setup: testSetup{
+				enableTCF1Fetch:    false,
+				enableTCF1Fallback: false,
+				vendorListVersion:  1,
+			},
+			expected: testExpected{
+				errorMessage: "gdpr vendor list version 1 does not exist, or has not been loaded yet. Try again in a few minutes",
+			},
+		},
+		{
+			description: "No Fetch - No Fallback - Vendor List 2",
+			setup: testSetup{
+				enableTCF1Fetch:    false,
+				enableTCF1Fallback: false,
+				vendorListVersion:  2,
+			},
+			expected: testExpected{
+				errorMessage: "gdpr vendor list version 2 does not exist, or has not been loaded yet. Try again in a few minutes",
+			},
+		},
+	}
+
+	for _, test := range testCases {
+		runTest(t, test, tcf1Version, server)
+	}
+}
+
+func TestTCF2FetcherInitialLoad(t *testing.T) {
+	// Loads two vendor lists during initialization by setting the latest vendor list version to 2.
+	// Ensures TCF1 fetch settings have no effect on TCF2.
+
+	server := httptest.NewServer(http.HandlerFunc(mockServer(serverSettings{
+		vendorListLatestVersion: 2,
+		vendorLists: map[int]string{
+			1: tcf2VendorList1,
+			2: tcf2VendorList2,
+		},
 	})))
 	defer server.Close()
 
-	fetcher := newVendorListFetcher(context.Background(), testConfig(), server.Client(), testURLMaker(server), 1)
-	list, err := fetcher(context.Background(), 2)
-	assertNilErr(t, err)
+	testCases := []test{
+		{
+			description: "Fetch - No Fallback - Vendor List 1",
+			setup: testSetup{
+				enableTCF1Fetch:    true,
+				enableTCF1Fallback: false,
+				vendorListVersion:  1,
+			},
+			expected: vendorList1Expected,
```

Comment:
I understand that. I'm using the common interface between both TCF1 and TCF2 for simplicity. The appropriate GVL  structures are created for TCF1 and TCF2 respectively.

It's not my intention to test the parser here, but rather to indirectly test it with basic success cases and focus on the logic of the vendor list fetcher. Since TCF1 is on it's way out, I think it might be better to make this specifically just a TCF2 fetcher in the future and then add a TCF3, etc.. fetcher as needed moving forward - rather than refactor it separately now. But I'm open to feedback.

Reply:
Yes, I don't have a better solution myself that doesn't involve one or more PRs, just wanted to pause for thought if there was a clearer way of signaling this.